### PR TITLE
Rename plot module to plotting

### PR DESCRIPTION
**Pull request by nvaytet**
We now import only the `plot` function from the `plotting` module in the scipp `__init__.py`.
This means we can now simply do `sc.plot(mydata)` instead of `sc.plot.plot(mydata)`.

The only other things in the `plotting` module that were being imported were some mostly unused shortcuts to various plot projections. These have now been removed.

This also fixes the mpl plots that were not showing in the binned_data notebook for example.

### CODE REVIEW

File: python/CMakeLists.txt [hunk 1]
```diff
@@ -50,29 +50,5 @@ endif()
...
-    scipp/src/plot/config.py
-    scipp/src/plot/dispatch.py
-    scipp/src/plot/plot_1d.py
-    scipp/src/plot/plot_2d.py
-    scipp/src/plot/plot_3d.py
-    scipp/src/plot/plot_collapse.py
-    scipp/src/plot/plot_matplotlib.py
-    scipp/src/plot/plot.py
-    scipp/src/plot/tools.py
-)
```
Comment (jl-wynen) [inline python/CMakeLists.txt:75]: Why did you remove them?

Comment (nvaytet) [reply python/CMakeLists.txt:75]: It seems that they are not used at all?
CMake apparently finds and copies all the files in the subdirectories.
See discussion on Slack (scipp channel)

**Comment by nvaytet** [comment]
@OwenArnold how would this impact the CI chain we have with the `ess` and `ess-notebooks` repos?
Am i right that they are now based on the release builds of Scipp, not `master`?
So when we make a new release of Scipp, we should update all the `ess` code and notebooks?

**Comment by OwenArnold** [comment]
> @OwenArnold how would this impact the CI chain we have with the `ess` and `ess-notebooks` repos?
> Am i right that they are now based on the release builds of Scipp, not `master`?
> So when we make a new release of Scipp, we should update all the `ess` code and notebooks?
> Will this lead to a potentially long build-up of errors/inconsistencies between Scipp releases?

https://github.com/scipp/scipp/issues/1494#issuecomment-778173167 describes current working. 

Yes correct. If you introduce breaking functionality in scipp, we don't see the effect downstream until the release is made. We could potentially have a `next`/`future`/`dev` branch on each of the repositories and also have pipelines that build from that using `scipp\label\dev`,

**Comment by SimonHeybrock** [comment]
*Do not merge right now* (converted to draft. This will break the IKON20 Python course, and I don't know when/if the images will be rebuilt.

**Comment by SimonHeybrock** [comment]
@nvaytet Or should we merge, fix the notebooks, and ask Torben to rebuild the image?

**Comment by nvaytet** [comment]
> @nvaytet Or should we merge, fix the notebooks, and ask Torben to rebuild the image?

I don't really mind. I'll ask Torben if there is still time to rebuild the image.
